### ISSUE TITLE
Unexpected end of JSON input

### ISSUE THREAD

**Issue by omeking2013** [pedr0fontoura/fivem-appearance#9]
![problem](https://user-images.githubusercontent.com/53650937/118085313-67cbdf00-b3ec-11eb-8d52-fa61c155a2fa.jpg)

Error when join the server and ui wont appear how to fix?? (1.1.0 version)

**Comment by MintKind** [comment]
Starting it the resource last solved this for me. Had it starting as very first before tho.

**Comment by mufty** [comment]
> Starting the resource last solved this for me. Had it starting as very first before tho.

Can confirm starting it last does help... otherwise same issue.

The modules I was starting after this one are:
start dpemotes
start RealisticVehicleFailure
start fh4speed

Just for context maybe they are in conflict.

**Comment by RockySouthpaw** [comment]
After asking someone about it they told me:

> It’s due to the fact the NUI inits before before game scripts.
```ts
on('onClientResourceStart', resourceName => {
  if (GetCurrentResourceName() != resourceName) {
    return;
  }
  init();
});
```
> The resource only initializes the NUI callbacks after the script is fully started.

Not sure if the fix would require a rework of the existing system but I don’t think starting the resource last is a reliable workaround tbh. 

Here’s a screenshot of the dev tools:

![](https://media.discordapp.net/attachments/795438488610537552/840432434944540752/unknown.png)

**Comment by mufty** [comment]
Yeah I just got the error even when starting it last

**Comment by Alessio-WAY** [comment]
Same here

**Comment by TzProwebyKS** [comment]
in resources.cfg

add start fivem-appearance
restart fivem-appearance

the error will go away

**Comment by ElSeba0811** [comment]
> After asking someone about it they told me:
> 
> > It’s due to the fact the NUI inits before before game scripts.
> 
> ```ts
> on('onClientResourceStart', resourceName => {
>   if (GetCurrentResourceName() != resourceName) {
>     return;
>   }
>   init();
> });
> ```
> 
> > The resource only initializes the NUI callbacks after the script is fully started.
> 
> Not sure if the fix would require a rework of the existing system but I don’t think starting the resource last is a reliable workaround tbh.
> 
> Here’s a screenshot of the dev tools:
> 
> ![](https://camo.githubusercontent.com/0f6c01c0dd51dd6a074bea2005891fa142153a345b78a9404b42c8efb45baedf/68747470733a2f2f6d656469612e646973636f72646170702e6e65742f6174746163686d656e74732f3739353433383438383631303533373535322f3834303433323433343934343534303735322f756e6b6e6f776e2e706e67)

I have also tried this but is not working, also tried putting the resource last and restart it, also doesnt work

**Comment by ZiggyJoJo** [comment]
Did anyone ever manage to fix this error?

**Comment by mufty** [comment]
I didn't for the moment I'm using this resource and starting it last that works most of the time while it still can crash sometimes. So I just need to watch out when starting the server for that. If what has been said here is true it's an easy fix tho just didn't have time myself to try to look into it and fix it for real.

**Comment by ZiggyJoJo** [comment]
> I didn't for the moment I'm using this resource and starting it last that works most of the time while it still can crash sometimes. So I just need to watch out when starting the server for that. If what has been said here is true it's an easy fix tho just didn't have time myself to try to look into it and fix it for real.

Yeah I tried that but it didn't work I mostly work with lua not js or HTML so I was hoping someone had an example of how that fix should be done otherwise I'm probably gonna spend the next few days banging my head against a wall till I figure it out

**Comment by snakewiz** [comment]
How I'm not able to reproduce this? I'm joining the server with the resource already running.

server.cfg

```cfg
ensure fivem-appearance
ensure test
```

test resource

```js
RegisterCommand(
  "customization",
  () => {
    globalThis.exports["fivem-appearance"].startPlayerCustomization((appearance) => {
      console.log(appearance);
    });
  },
  false
);
```

**Comment by ZiggyJoJo** [comment]
`setr fivem-appearance:locale "en"
ensure fivem-appearance`

`RegisterCommand('customization', function()
	local config = {
		ped = true,
		headBlend = true,
		faceFeatures = true,
		headOverlays = true,
		components = true,
		props = true
	}

	Citizen.Wait(1000)

  
	exports['fivem-appearance']:startPlayerCustomization(function (appearance)
		if (appearance) then
			TriggerServerEvent('fivem-appearance:save', appearance)
			print('Saved')
		else
			print('Canceled')
		end
	end, config)
end, false)`

This is how I've got it and it throw's this error on login

![image](https://user-images.githubusercontent.com/69662456/121461319-9ce03880-c97c-11eb-987b-b1bebd7d0dd4.png)

**Comment by pedr0fontoura** [comment]
Is this fixed after c7458d259633b7fbc31d2d098b13ef0d07ed5829?
